### PR TITLE
y_vehicledata's new friend

### PR DESCRIPTION
**Pull request by DEntis-T**

### CODE REVIEW

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    88  `8b   d8'  88 ""     `Y8 88 88P'   `"8a       d8YaaaaY8b    88""""""'   88  
+    88   `8b d8'   88 ,adPPPPP88 88 88       88      d8""""""""8b   88          88  
+    88    `888'    88 88,    ,88 88 88       88     d8'        `8b  88          88  
+    88     `8'     88 `"8bbdP"Y8 88 88       88    d8'          `8b 88          88  
+
+*/
+
+enum E_WEAPON_DATA
+{
+    Name[24],
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: Why are the names needed here? There's already `GetWeaponName`, which does that (and returns different values for some weapons - we should keep consistency).

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    ID,
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: And why is the ID needed?  It matches the array index, so you can just return that.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    Slot,
+    Model
+}
+
+new YSI_g_WeaponInformation[47][E_WEAPON_DATA] = {
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: `static stock`.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    88`8b       d8'88                                   d8'`8b      88      ,8P 88  
+    88 `8b     d8' 88 ,adPPYYba, 88 8b,dPPYba,         d8'  `8b     88aaaaaa8P' 88  
+    88  `8b   d8'  88 ""     `Y8 88 88P'   `"8a       d8YaaaaY8b    88""""""'   88  
+    88   `8b d8'   88 ,adPPPPP88 88 88       88      d8""""""""8b   88          88  
+    88    `888'    88 88,    ,88 88 88       88     d8'        `8b  88          88  
+    88     `8'     88 `"8bbdP"Y8 88 88       88    d8'          `8b 88          88  
+
+*/
+
+enum E_WEAPON_DATA
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: This should be multiple separate arrays, not an enum array.  E.g. an array for slots, and a separate array for models.  This allows the compiler to better elide unused data.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+    {"Night Vis Goggles", 44, 11, 368 },
+    {"Thermal Goggles", 45, 11, 369 },
+    {"Parachute", 46, 11, 371 }
+};
+
+//P:D(Weapon_IsValid(weaponid));
+//#define Weapon_IsValid(%0) (%0>=1 && %0<=18 || %0>=21 && %0<=46)
+stock bool:Weapon_IsValid(weaponid)
+{
+	return (weaponid >= 1 && weaponid <= 18 || weaponid >= 21 && weaponid <= 46) ? (true) : (false);
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: You don't need the ternary, just return the expression.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 6]
```diff
@@ -0,0 +1,213 @@
...
+}
+
+stock bool:Weapon_IsAssaultRifle(weaponid)
+{
+	switch(weaponid)
+	{
+		case 30, 31: return true;
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc:507]: Or `slot == 5`?

**Review by Y-Less — CHANGES_REQUESTED** [review]
I appreciate the library, and how well it is done in terms of matching other YSI libraries.  There is still some things to improve, but I like where it is going.  Some other features to consider:

* Ammo information.
* weaponconf.dat information.
* Related weapon skill values.
* Usage animations.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc (hunk 6 above)
Comment (DEntis-T) [reply YSI_Game/y_weapondata/y_weapondata_entry.inc:507]: I need to check the slot of the weapon?

**Comment by DEntis-T** [comment]
Added the function you mentioned on Discord.

**Comment by Y-Less** [comment]
Another idea (I'm just spewing them out, I don't know if they are good) - weapon ID from model ID.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc (hunk 6 above)
Comment (Y-Less) [reply YSI_Game/y_weapondata/y_weapondata_entry.inc:507]: I mean, isn't that an easier way to check the type?

**Comment by DEntis-T** [comment]
I've already done that.

**Comment by DEntis-T** [comment]
Check the last commit.

**Comment by xunder-matth** [comment]
`Weapon_GetAmmoName` and `Weapon_GetAnimGroup` should be `Weapon_Return(...)`.

Examples:

https://github.com/pawn-lang/YSI-Includes/blob/d91cdefa63789c97ae6abde016dc6b66c155b4bc/YSI_Core/y_core/y_utils_impl.inc#L1847-L1853

https://github.com/pawn-lang/YSI-Includes/blob/d91cdefa63789c97ae6abde016dc6b66c155b4bc/YSI_Core/y_core/y_utils_impl.inc#L1865-L1878

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 7]
```diff
@@ -0,0 +1,511 @@
...
+	10.1, //{"Fire Extinguisher", 42, 9, 366 },
+	100.0, //{"Camera", 43, 9, 367 },
+	100.0, //{"Night Vis Goggles", 44, 11, 368 },
+	100.0, //{"Thermal Goggles", 45, 11, 369 },
+	25.0  //{"Parachute", 46, 11, 371 }
+};
+
+stock Float:Weapon_GetMaxRange(weaponid)
+{
+	return Weapon_IsValid(weaponid) ? YSI_g_MaxWeaponRange[weaponid] : (-1);
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc:317]: Still using magic numbers.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 8]
```diff
@@ -0,0 +1,511 @@
...
+	WEAPONDATA_NO_AMMO, //{"Fire Extinguisher", 42, 9, 366 },
+	WEAPONDATA_NO_AMMO, //{"Camera", 43, 9, 367 },
+	WEAPONDATA_NO_AMMO, //{"Night Vis Goggles", 44, 11, 368 },
+	WEAPONDATA_NO_AMMO, //{"Thermal Goggles", 45, 11, 369 },
+	WEAPONDATA_NO_AMMO  //{"Parachute", 46, 11, 371 }
+};
+
+stock Weapon_GetAmmoName(weaponid)
+{
+	new dest[YSI_WEAPDATA_MAX_STRING_SIZE];
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: Don't return strings, pass by reference.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 9]
```diff
@@ -0,0 +1,511 @@
...
+	if(!Weapon_IsValid(weaponid))
+	{
+		strcat(dest, " ");
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: Just return a blank string, why a space?

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 10]
```diff
@@ -0,0 +1,511 @@
...
+stock Weapon_GetAnimGroup(weaponid)
+{
+	new dest[YSI_WEAPDATA_MAX_STRING_SIZE];
+	if(!Weapon_IsValid(weaponid))
+	{
+		strcat(dest, " ");
+		return dest;
+	}
+	else strcat(dest, YSI_g_WeaponAnimGroups[weaponid]);
+	return dest;
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: Same as above on strings.

**Review by Y-Less — CHANGES_REQUESTED** [review]

**Comment by Y-Less** [comment]
What is the status of this @DEntis-T ?

**Comment by DEntis-T** [comment]
I was on a long break, I will finish it soon

**Comment by DEntis-T** [comment]
@Y-Less I think that this can be merged, I am out of ideas for more features now.

**Pull request merged**